apps/video_edit/__init__.py

```diff
@@ -0,0 +1,154 @@
+import superdesk
+from apps.archive.common import ARCHIVE
+from superdesk import config
+from superdesk.media.video_editor import VideoEditorWrapper
+from superdesk.metadata.utils import item_url
+
+
+class VideoEditService(superdesk.Service):
+    """
+    Use video server for editing video.
+    """
+
+    video_editor = VideoEditorWrapper()
+
+    def create(self, docs, **kwargs):
+        ids = []
+        for doc in docs:
+            item = doc.get('item')
+            item_id = item.get(config.ID_FIELD)
+            media_id = item['media']
+            renditions = item.get('renditions', {})
+            # push task capture preview thumbnail to video server
+            if 'capture' in doc:
+                # Remove empty value in updates to avoid cerberus return invalid input error
+                capture = doc.pop('capture')
+                if capture:
+                    project = self.video_editor.capture_preview_thumbnail(
+                        media_id, position=capture.get('position'),
+                        crop=capture.get('crop'),
+                        rotate=capture.get('rotate')
+                    )
+                    renditions.setdefault('thumbnail', {}).update({
+                        'href': project['thumbnails']['preview'].get('url'),
+                        'mimetype': project['thumbnails']['preview'].get('mime_type', 'image/png'),
+                    })
+            # push task edit video to video server
+            if 'edit' in doc:
+                edit = doc.pop('edit')
+                if edit:
+                    project = self.video_editor.edit(media_id, edit)
+                    media_id = project.get('_id')
+                    renditions.setdefault('original', {}).update({
+                        'href': project['url'],
+                        'version': project['version'] + 1,
+                        'video_editor_id': media_id,
+                    })
+            if renditions:
+                updates = self.patch(item_id, {
+                    'renditions': renditions,
+                    'media': media_id,
+                })
+                item.update(updates)
+            ids.append(item_id)
+        return ids
+
+    def find_one(self, req, **lookup):
+        res = super().find_one(req, **lookup)
+        if req is None:
+            return res
+
+        video_id = res['media']
+        if req.args.get('action') == 'timeline':
+            response = self.video_editor.create_timeline_thumbnails(video_id, req.args.get('amount', 40))
+            return {
+                config.ID_FIELD: video_id,
+                **response
```

Comment:
This will raise an error. In this case, `response` is a list.

Reply:
Hi, Oleg,
Should we only update response of timeline from video server ([change](https://github.com/superdesk/video-server/pull/19/files))?
It make format of response is homogeneous, easier to handling.

Reply:
@idsvn I'm not sure If I understood. In case when action is `timeline` it raises an error.